### ISSUE TITLE
Dash Studio - Repetitions and visibility

### ISSUE THREAD

**Issue by Romainrob** [SHWotever/SimHub#593]
Hi.
I just noticed on groups with repetition, when disabling visibility the formulas inside seem to still be running : https://youtu.be/sWyDz6Yz6s8
Is this the expected behavior?
Bonne soirée.

**Comment by Romainrob** [comment]
Hi,
Yesterday I realize the "issue" actually seems to be that the visibility inherited from a layer doesn't stop the fx of the components inside it from running.
![image](https://user-images.githubusercontent.com/50543913/98935077-46447600-24e3-11eb-94ff-e6efbdab3384.png)
Is this normal?
Bonne journée.

**Comment by Romainrob** [comment]
Hi,
Looking into this again as I noticed I get very different performance numbers when running my heaviest dash in HTML and desktop mode. Here is a comparative test with a paused replay.
HTML:
![HTML MODE](https://user-images.githubusercontent.com/50543913/121607139-cfa73100-ca4f-11eb-8f95-c2cca891472b.png)

Desktop:
![DESKTOP MODE](https://user-images.githubusercontent.com/50543913/121607173-dd5cb680-ca4f-11eb-945e-b607f6299243.png)

I suspect in HTML it's constantly running all the invisible stuff, and there is a lot of it as it has a page system (based on layer visibility, not widget pages).
I remember you explained all the formulas run once upon loading in HTML mode. So I expected a similar load on the formula engine after loading. It doesn't seem to be the case.

So I went back to testing if the FX of invisible components where constantly running, using an FX that would throw an error on purpose.
And looking at the logs, Invisible FX are indeed running, but the same in HTML and desktop mode. Which doesn't explained why I get this performance difference. Unless the fact that it throws an errors makes this test irrelevant somehow?

![Logged error Test dash](https://user-images.githubusercontent.com/50543913/121607866-1e08ff80-ca51-11eb-8c85-4a0d6413264c.png)

So I thought I was on to something but maybe not.
Idk if this point an issue an potential performance gains. I'll let you be the judge of that.

Here I included everything I used for these tests:
https://www.dropbox.com/s/ses4yvwt1jdr68p/DS%20Insisible%20FX%20test.zip?dl=1

Please let me now your thoughts on this as I'm quite confused now.
Bonne nuit/journée en fonction :)

**Comment by Romainrob** [comment]
Hi again. I noticed my mistake after the fact on the "logged errors" test. I forgot to hide the "invisible error" component.
Now I can see the Fx from the invisible component doesn't run, when the one from the component inheriting invisibility from the layer does.

![Logged error Test dash2](https://user-images.githubusercontent.com/50543913/121610504-7f7f9d00-ca56-11eb-9ed2-89a6ff6f86a7.png)

So this is consistent with what I noticed before but both HTML and desktop mode are still behaving the same.
Yes, I'm still confused... :)
Allez, bonne nuit cette fois.
